modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/exp/ExpressionFactoryImpl.java

```diff
@@ -343,6 +345,17 @@ public Iterable<RowT> values(List<RexLiteral> values, RelDataType rowType) {
         return rows;
     }
 
+    private @Nullable Object literalValue(RexLiteral literal, Class<?> type) {
+        Object val = literal.getValueAs(type);
+
+        // Literal was parsed as UTC timestamp, now we need to adjust it to the client's time zone.
+        if (val != null && literal.getTypeName() == SqlTypeName.TIMESTAMP_WITH_LOCAL_TIME_ZONE) {
```

Comment:
Should we adjust `TIME_WITH_LOCAL_TIME_ZONE` as well?

Reply:
we have separate ticket for this https://issues.apache.org/jira/browse/IGNITE-21555